saltstack/base/salt/luks/etc/systemd/system/reopen-luks-volume.service

```diff
@@ -0,0 +1,19 @@
+[Unit]
+Description=LUKS volume reopen script service
+ConditionPathIsMountPoint=!/mnt/luks
+ConditionFileNotEmpty=/etc/luks/passphrase_encryption_key
+ConditionFileNotEmpty=/etc/luks/passphrase_ciphertext
+After=cloud-init.target
```

Comment:
```sls
start-cloudera-scm-server:
  service.running:
    - name: cloudera-scm-server
```
I guess a salt highstate starts it after a reboot.

Reply:
I changed the reopen service to autostart after `network-online.target` is reached as it is indeed the point where the network can actually considered be "up" (see: https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/), and we should be able to decrypt the passphrase ciphertext. 

This would mean it would start in parallel with the CM service for example, which is not what we want, so I added a TODO and a note to the unit file, to add a `Before` directive for services like the CM service, to ensure the LUKS volume is mounted before these services would start and try to use secrets stored on the encrypted volume.
I think adding these lines to the unit file can be done when we are actually going through the secrets and creating the symlinks for them.

I assume no service is trying to start before the network is "up", as in that case I do not know what we should do 👀

Reply:
Great. I agree we should somehow make sure that `cdp-reopen-luks-volume` starts and completes before `cloudera-scm-server`.

Since we are currently focusing on getting FreeIPA up & running, I suggest raising a task for this under the epic and have it handled later when we get to dealing with DL & DH. Could you please create a Jira ticket for this? :smile: